### DISCUSSION TITLE
Get AlsacoCtrl running on a Pi to use with IOBroker

### DISCUSSION THREAD

**Discussion by Vippis2000** [strandborg/AlsavoCtrl#24] [General]
Hi strandborg,

I own a Brilix/Albixon Inverboost XHPFD PX100 Pool Heatpump and want to monitor and control it via IOBroker, running on a Raspberry Pi.

I want to run your tool on the Pi and forward the status to IOBroker. IOBroker offers a MQTT Broker. 

Can you give me a little help how to set the system up?

1. How do I get AlsavoCtrl running on the Pi?
2. How do I setup MQTT?
3. How shall I setup MQTT Broker in IOBroker?

Thanks a lot!
Alex

**Comment by strandborg** [comment]
I think the easiest way is to use it via Docker and the included Dockerfile; alternatively, you can follow the build steps in Dockerfile to roll your own binary on Raspberry PI (that was my approach before I migrated to Docker). As for MQTT broker, a basic MQTT broker setup will do, just feed the broker IP, user and password to alsavo.py script.

**Comment by Vippis2000** [comment]
Thank you I'll try i...

Can I also replace Alsavo Ip and UDP port in the python file to have local communication to the pump ?

**Comment by strandborg** [comment]
yes, you can indeed. Sometimes it works more reliably that way, sometimes the other way around.

**Comment by Vippis2000** [comment]
The manual says that the pump has either WiFi or Modbus control. Do you if the control board has a modbus port or is it on a separate control board as the WiFi unit ?

**Comment by strandborg** [comment]
I don't have any information about the modbus connection. If you find out, please let me know!

**Comment by Vippis2000** [comment]
Hi again, I have installed AlsavoCtrl using Docker on my PI and could read sensor information via MQTT in IOBroker. But I'm not able to set values, for example heatsetpoint. How do I do this via MQTT?

**Reply by Vippis2000** [reply]
If i change the state of "set" in heatingsetpoint to e.g. "29" nothing happens

**Reply by strandborg** [reply]
Hmm this should work - I am using OpenHAB and Mosquitto MQTT broker and setting the target temperature works like you specified. Are there any errors in the logs?

**Reply by Vippis2000** [reply]
No there are no errors in the log

With the same instance of MQTT broker in IOBroker I'm getting sensor information from Tasmota NeoPool from my Hydrolysis device. There I'm able to send commands to a certain datapoint with in a cmnd folder...

**Comment by strandborg** [comment]
You could try an external mqtt tool such as MQTT Explorer, maybe there’s a
separate topic for setting the heatpoint vs reading its current state. This
part is handled by the Homie4 lib so I don’t remember how it’s set out off
the top of my head.

**Comment by Vippis2000** [comment]
Got it all running now, had to switch to a separate MQTT adapter instance...

**Reply by Vippis2000** [reply]
@strandborg After 3 days of constant running I can't set values any more. Same behaviour as before. How can I see the log file in the docker version?

**Reply by Vippis2000** [reply]
Only by restarting the MQTT broker he once sets the value in the set topic, but after that no setting will be accepted.

**Reply by strandborg** [reply]
This depends on the docker system you're running. For example, in Portainer you can see logs directly from the UI.

**Reply by Vippis2000** [reply]
I'm using standard docker on my PI

**Reply by strandborg** [reply]
https://docs.docker.com/reference/cli/docker/container/logs/

**Reply by Vippis2000** [reply]
![grafik](https://github.com/strandborg/AlsavoCtrl/assets/128602940/0c6b7262-19fa-4ebb-89e1-2bf5c27cb724)

**Reply by Vippis2000** [reply]
This is the log output when I send "false" in power set topic

**Reply by Vippis2000** [reply]
but pump is still active as I can see on alsavo pro

**Reply by strandborg** [reply]
I guess something is going wrong with MQTT broker... No idea what, though.

**Reply by Vippis2000** [reply]
I'll ask IOBroker MQTT Broker devs, thanks!

**Reply by Vippis2000** [reply]
MQTT Broker needs to send messages without Retain Flag, then commands work. 

JS: sendTo('mqtt.1', 'sendMessage2Client', { topic: 'homie/alsavo/controls/power/set', message: 'false', retain: false });
